rest/bootstrap_test.go

```diff
@@ -131,6 +131,46 @@ func TestBootstrapRESTAPISetup(t *testing.T) {
 	assertResp(t, resp, http.StatusOK, `{"_id":"doc1","_rev":"1-cd809becc169215072fd567eebd8b8de","foo":"bar"}`)
 }
 
+// TestBootstrapDuplicateBucket will attempt to create two databases sharing the same bucket and ensure this isn't allowed.
+func TestBootstrapDuplicateBucket(t *testing.T) {
+	if base.UnitTestUrlIsWalrus() {
+		t.Skip("Bootstrap works with Couchbase Server only")
+	}
+
+	defer base.SetUpTestLogging(base.LevelInfo, base.KeyHTTP)()
+
+	// Start SG with no databases in bucket(s)
+	config := bootstrapStartupConfigForTest(t)
+	sc, err := setupServerContext(&config, true)
+	require.NoError(t, err)
+	serverErr := make(chan error, 0)
+	go func() {
+		serverErr <- startServer(&config, sc)
+	}()
+	require.NoError(t, sc.waitForRESTAPIs())
+
+	// Get a test bucket, and use it to create the database.
+	tb := base.GetTestBucket(t)
+	defer func() { tb.Close() }()
+	resp := bootstrapAdminRequest(t, http.MethodPut, "/db1/",
+		`{"bucket": "`+tb.GetName()+`", "num_index_replicas": 0}`,
```

Comment:
Would be nice to add the use views / xattrs config options based on the test vars